index.js

```diff
@@ -48,12 +49,50 @@ module.exports = {
       ssb.metafeeds.findOrCreate(details, cb)
     }
 
-    function findOrCreateGroupFeed(input, cb) {
-      const secret = Buffer.isBuffer(input) ? new SecretKey(input) : input
+    function findEmptyGroupFeed(rootId, cb) {
+      let found = false
+      pull(
+        ssb.metafeeds.branchStream({ root: rootId, old: true, live: false }),
+        pull.filter((branch) => branch.length === 4),
+        pull.map((branch) => branch[3]),
+        // only grab feeds that look like group feeds
+        pull.filter((feed) => feed.recps && feed.purpose.length === 44),
+        paraMap((feed, cb) => {
+          return pull(
+            ssb.db.query(where(author(feed.id)), toPullStream()),
+            pull.take(1),
+            pull.collect((err, res) => {
+              if (res.length === 0) {
+                // we're only interested in empty feeds
+                return cb(null, feed)
+              } else {
+                return cb()
+              }
+            })
+          )
```

Comment:
There's a better way to count the size of a feed that doesn't depend on actually retrieving the messages from the log (thus relying only on jitdb indexes) and without pull.collect:

```js
ssb.db.query(where(author(feed.id)), count(), toCallback(cb)
```

The `cb` is called as `cb(err, num)` where `num` is the number of messages matching the query. Then in the filter you can do `pull.filter(num => num === 0)`.

Well, it's not going to be that simple because you then have `pull.unique('id')`, but at least this gives a hint how to get it done

Reply:
done